spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala

```diff
@@ -246,6 +280,47 @@ object TableFeature {
   /** Get a [[TableFeature]] object by its name. */
   def featureNameToFeature(featureName: String): Option[TableFeature] =
     allSupportedFeaturesMap.get(featureName.toLowerCase(Locale.ROOT))
+
+  /**
+   * Extracts the removed feature name by comparing current and base protocols.
+   * Returns None if there is no removed feature.
+   * Assumes there can only be one feature removed at a time.
+   */
+  protected def getRemovedFeatureName(
+      currentProtocol: Protocol,
+      baseProtocol: Protocol): Option[String] = {
+    val currentFeatureNames = currentProtocol.readerAndWriterFeatureNames
+    val baseFeatureNames = baseProtocol.readerAndWriterFeatureNames
+    val removedFeatures = baseFeatureNames -- currentFeatureNames
+
+    require(removedFeatures.size <= 1)
+    Option(baseFeatureNames -- currentFeatureNames).filter(_.nonEmpty).map(_.head)
+  }
+
+  /**
+   * Identifies whether there was a feature removal between two protocols.
+   */
+  def isDropFeatureCommit(currentProtocol: Protocol, baseProtocol: Protocol): Boolean = {
```

Comment:
We're not comparing commits anymore. Let's name this `isProtocolRemovingFeatures`. Also, the name `currentProtocol` is not really correct -- it kind of echoes the origin of this function's usage in conflict checking. But really all this function needs to know which of the protocols is the "before" and which is the "after". So then you get `isProtocolRemovingFeatures(newProtocol, oldProtocol)`.